resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -0,0 +1,376 @@
+    RomanceStandardTime
+    CentralEuropeanStandardTime
+    WCentralAfricaStandardTime
+    GTBStandardTime
+    MiddleEastStandardTime
+    EgyptStandardTime
+    EEuropeStandardTime
+    WestBankStandardTime
+    SouthAfricaStandardTime
+    FLEStandardTime
+    IsraelStandardTime
+    SouthSudanStandardTime
+    KaliningradStandardTime
+    SudanStandardTime
+    LibyaStandardTime
+    NamibiaStandardTime
+    JordanStandardTime
+    ArabicStandardTime
+    SyriaStandardTime
+    TurkeyStandardTime
+    ArabStandardTime
+    BelarusStandardTime
+    RussianStandardTime
+    EAfricaStandardTime
+    VolgogradStandardTime
+    IranStandardTime
+    ArabianStandardTime
+    AstrakhanStandardTime
+    AzerbaijanStandardTime
+    RussiaTimeZone3
+    MauritiusStandardTime
+    SaratovStandardTime
+    GeorgianStandardTime
+    CaucasusStandardTime
+    AfghanistanStandardTime
+    WestAsiaStandardTime
+    QyzylordaStandardTime
+    EkaterinburgStandardTime
+    PakistanStandardTime
+    IndiaStandardTime
+    SriLankaStandardTime
+    NepalStandardTime
+    CentralAsiaStandardTime
+    BangladeshStandardTime
+    OmskStandardTime
+    MyanmarStandardTime
+    SEAsiaStandardTime
+    AltaiStandardTime
+    WMongoliaStandardTime
+    NorthAsiaStandardTime
+    NCentralAsiaStandardTime
+    TomskStandardTime
+    ChinaStandardTime
+    NorthAsiaEastStandardTime
+    SingaporeStandardTime
+    WAustraliaStandardTime
+    TaipeiStandardTime
+    UlaanbaatarStandardTime
+    AusCentralWStandardTime
+    TransbaikalStandardTime
+    TokyoStandardTime
+    NorthKoreaStandardTime
+    KoreaStandardTime
+    YakutskStandardTime
+    CenAustraliaStandardTime
+    AUSCentralStandardTime
+    EAustraliaStandardTime
+    AUSEasternStandardTime
+    WestPacificStandardTime
+    TasmaniaStandardTime
+    VladivostokStandardTime
+    LordHoweStandardTime
+    BougainvilleStandardTime
+    RussiaTimeZone10
+    MagadanStandardTime
+    NorfolkStandardTime
+    SakhalinStandardTime
+    CentralPacificStandardTime
+    RussiaTimeZone11
+    NewZealandStandardTime
+    UTC12
+    FijiStandardTime
+    KamchatkaStandardTime
+    ChathamIslandsStandardTime
+    UTC13
+    TongaStandardTime
+    SamoaStandardTime
+    LineIslandsStandardTime
+
+}
+
+#endRegion Enums
+function Get-ValidTimeZone {
+    param (
+        [Parameter()]
+        # keep it string to avoid enum issues
+        [string] $TimeZone = ((Get-TimeZone).Id -replace '[\+\s\-\(\)\.]', ''),
+
+        # switch for Get() method
+        [Parameter()]
+        [switch] $NoValid
+    )
+
+    $list = (Get-TimeZone -ListAvailable).Id
+
+    $trimmedVersion = $list -replace '[\+\s\-\(\)\.]', ''
+
+    if ($trimmedVersion -contains $TimeZone) {
+        if ($NoValid.IsPresent) {
+            return $TimeZone
+        }
+
+        return $list[$trimmedVersion.IndexOf($TimeZone)]
+    } else {
+        throw 'Invalid time zone. Please provide a valid time zone without spaces and special characters.'
+    }
+}
```

Comment:
This can be simplified by just checking if the timezoneId provided is valid by calling `Get-TimeZone -Id <timezoneid>` and seeing if that returns a valid response.

Reply:
Done! The effect without it, is that users don't get the IntelliSense. Docs should guide them.